File: app/src/main/java/ru/systempla/weatherapp/di/module/RepoModule.java
package ru.systempla.weatherapp.di.module;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import ru.systempla.weatherapp.mvp.model.api.IDataSource;
import ru.systempla.weatherapp.mvp.model.repo.IWeatherRepo;
import ru.systempla.weatherapp.mvp.model.repo.WeatherRepo;


@Module(includes = {ApiModule.class})
public class RepoModule {

    @Singleton
    @Provides
    public IWeatherRepo weatherRepo(IDataSource api) {
        return new WeatherRepo(api);
    }
}

File: app/src/main/java/ru/systempla/weatherapp/di/module/RepoModule.kt
package ru.systempla.weatherapp.di.module

import dagger.Module
import dagger.Provides
import ru.systempla.weatherapp.mvp.model.api.IDataSource
import ru.systempla.weatherapp.mvp.model.repo.IWeatherRepo
import ru.systempla.weatherapp.mvp.model.repo.WeatherRepo
import javax.inject.Singleton

@Module(includes = [ApiModule::class])
class RepoModule {
    @Singleton
    @Provides
    fun weatherRepo(api: IDataSource): IWeatherRepo = WeatherRepo(api)
}